test/client/app/components/addSite/addSite.test.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { expect } from 'chai';
+import { stub, spy } from 'sinon';
+import templates from '../../../../../config/templates';
+import proxyquire from 'proxyquire';
+
+proxyquire.noCallThru();
+
+const mock = () => {
+  return function() {
+    return <div></div>;
+  };
+};
+
+const TemplateSiteList = mock();
+const LinkButton = mock();
+const AlertBanner = mock();
+
+const cloneRepo = spy();
+const addSite = stub();
+
+const user = {
+  username: 'jill',
+  id: '1'
+};
+
+const error = {error: 'whoooooops'};
+
+const propsWithoutError = {
+  storeState: { user, error: ''}
+};
+
+const Fixture = proxyquire('../../../../../assets/app/components/AddSite', {
+  './TemplateSiteList': TemplateSiteList,
+  '../linkButton': LinkButton,
+  '../alertBanner': AlertBanner,
+  '../../actions/siteActions': { cloneRepo, addSite  },
+  '../../../../config/templates': templates
+}).default;
+
+describe('<AddSite/>', () => {
+  let wrapper;
+
+  beforeEach(() => {
+    wrapper = shallow(<Fixture {...propsWithoutError} />);
+  });
+
+  it('has expected default state based on supplied props', () => {
+    const actualState = wrapper.state();
+    const expectedState = {
+      owner: user.username,
+      users: [+user.id],
+      engine: 'jekyll',
+      defaultBranch: 'master',
+      repository: ''
+    };
+
+    expect(actualState).to.deep.equal(expectedState);
+  });
+
+  it('renders form input elements with the correct initial values', () => {
+    const { owner, repository, engine, defaultBranch } = wrapper.state();
+
+    expect(wrapper.find('input[name="owner"]').props().value).to.equal(owner);
+    expect(wrapper.find('input[name="repository"]').props().value).to.equal(repository);
+    expect(wrapper.find('select[name="engine"]').props().value).to.equal(engine);
+    expect(wrapper.find('input[name="defaultBranch"]').props().value).to.equal(defaultBranch);
+  });
+
+  it('renders its children', () => {
+    expect(wrapper.find(TemplateSiteList)).to.have.length(1);
+    expect(wrapper.find(AlertBanner)).to.have.length(1);
+    expect(wrapper.find(LinkButton)).to.have.length(1);
+    expect(wrapper.find('form')).to.have.length(1);
+  });
+
+  it('calls the clone repo action when a template is selected', () => {
+    const templateRepo = 'app';
+    const siteName = 'microsite';
+
+    wrapper.instance().onSubmitTemplate(templateRepo, siteName);
+
+    expect(cloneRepo.calledWith({repo: templateRepo}, templates[siteName])).to.be.true;
+    expect(cloneRepo.called).to.be.true;
+  });
+
+  it('delivers the correct props to its children', () => {
+    const bannerProps = wrapper.find(AlertBanner).props();
+    const templateListProps = wrapper.find(TemplateSiteList).props();
+    const linkButtonProps = wrapper.find(LinkButton).props();
```

Comment:
Is the intent to test if the right props are given to the link button? These props aren't validated in the test.